lua/nvim-surround/buffer.lua

```diff
@@ -117,19 +117,7 @@ end
 ---@param stop integer The final line.
 ---@param lines string[] The set of lines to replace the lines in the buffer.
 M.set_lines = function(start, stop, lines)
-    vim.api.nvim_buf_set_lines(0, start - 1, stop, false, lines)
-end
-
--- Inserts a set of lines into the buffer at a given position.
----@param pos integer[] The position to be inserted at.
----@param lines string[] The lines to be inserted.
-M.insert_lines = function(pos, lines)
-    local line = pos[1] > #M.get_lines(1, -1) and "" or M.get_line(pos[1])
-    -- Make a copy of the lines to avoid modifying delimiters
-    local lns = vim.deepcopy(lines)
-    lns[#lns] = lns[#lns] .. line:sub(pos[2])
-    lns[1] = line:sub(1, pos[2] - 1) .. lns[1]
-    M.set_lines(pos[1], pos[1], lns)
+    vim.api.nvim_buf_set_text(0, start - 1, 0, stop - 1, #M.get_line(stop), lines)
```

Comment:
This basically just replicates `nvim_buf_set_lines()` in terms of `nvim_buf_set_text()` and doesn't gain you anything.

Per `:h nvim_buf_set_text()`:
>This is recommended over nvim_buf_set_lines when only modifying parts of a line, as extmarks will be preserved on non-modified parts of the touched lines.

>Prefer nvim_buf_set_lines when adding or deleting entire lines only.

However, I don't think there's anything this plugin does which needs to replace entire lines at all. Most if not all operations should be done in terms of two calls to `nvim_buf_set_text()`, one for the opening delimiter and one for the closing delimiter.

Reply:
Yeah, I just realized this and have removed the function altogether.